torch/csrc/jit/script/compiler.cpp

```diff
@@ -2376,17 +2398,35 @@ struct to_ir {
       at::ArrayRef<NamedValue> inputs,
       at::ArrayRef<NamedValue> attributes) {
     // Build the fork node without inputs
-    auto fork_node =
-        method.graph()
-            ->insertNode(method.graph()->create(prim::fork, 1))
-            ->setSourceRange(loc);
+    auto fork_node = method.graph()
+                         ->insertNode(method.graph()->create(prim::fork, 1))
+                         ->setSourceRange(loc);
     auto body_block = fork_node->addBlock();
 
     // Build a template of the graph to be executed
     Value* node_output;
     {
       WithInsertPoint guard(body_block);
-      auto fn_sugared_output = forked->call(loc, method, inputs, attributes, 1);
+
```

Comment:
This patch is suppose to be a cleanup patch, adding additional workarounds is not going to work. Consider just inserting a CallClosure primitive, which is then removed in a cleanup pass. It is closer to how this should eventually work, and doesn't require hacks.